lib/mailtrap/mail.rb

```diff
@@ -50,10 +53,19 @@ def from_message(message) # rubocop:disable Metrics/AbcSize, Metrics/MethodLengt
 
       HEADERS_TO_REMOVE = (SPECIAL_HEADERS + ACTIONMAILER_ADDED_HEADERS).freeze
 
+      # @param header [Mail::Field, nil]
+      # @return [Mail::AddressList, nil]
       def address_list(header)
-        header.respond_to?(:element) ? header.element : header&.address_list
+        return nil unless header
+
+        unless header.errors.empty?
```

Comment:
interestingly the one word email address e.g. `invalid` is parsed with no issues. Server then returns "invalid email address" error.